storage/statedb/sync_bloom.go

```diff
@@ -94,32 +94,39 @@ func NewSyncBloom(memory uint64, database database.Iteratee) *SyncBloom {
 }
 
 // init iterates over the database, pushing every trie hash into the bloom filter.
-func (b *SyncBloom) init(database database.Iteratee) {
+func (b *SyncBloom) init(db database.Iteratee) {
 	// Iterate over the database, but restart every now and again to avoid holding
 	// a persistent snapshot since fast sync can push a ton of data concurrently,
 	// bloating the disk.
 	//
 	// Note, this is fine, because everything inserted into leveldb by fast sync is
 	// also pushed into the bloom directly, so we're not missing anything when the
 	// iterator is swapped out for a new one.
-	it := database.NewIterator(nil, nil)
+	it := db.NewIterator(nil, nil)
 
 	var (
 		start = time.Now()
 		swap  = time.Now()
 	)
 	for it.Next() && atomic.LoadUint32(&b.closed) == 0 {
 		// If the database entry is a trie node, add it to the bloom
-		if key := it.Key(); len(key) == common.HashLength {
+		key := it.Key()
+		if len(key) == common.HashLength {
 			b.bloom.Add(syncBloomHasher(key))
 			bloomLoadMeter.Mark(1)
 		}
+		// If the database entry is a contract code, add it to the bloom
+		if ok, hash := database.IsCodeKey(key); ok {
```

Comment:
@kjhman21 There is no case that you mentioned. The length of code key should be `len(key) == common.HashLength+len(codePrefix)`. This condition is checked in `IsCodeKey` method.

Reply:
Yeah, I have no objection to merge this as it is, but if the conditions cannot be met both cases, it would be better to add 'else' statement like

```
if len(key) == common.HashLength {
} else if ok, hash := database.IsCodeKey(key); ok {
```